ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -29,6 +29,66 @@ def swish(input_activation: torch.Tensor) -> torch.Tensor:
         """Swish activation function. For more info: https://arxiv.org/abs/1710.05941"""
         return torch.mul(input_activation, torch.sigmoid(input_activation))
 
+    @staticmethod
+    def apply_learning_rate(optim: torch.optim.Optimizer, lr: float) -> None:
+        """
+        Apply a learning rate to a torch optimizer.
+        :param optim: Optimizer
+        :param lr: Learning rate
+        """
+        for param_group in optim.param_groups:
+            param_group["lr"] = lr
+
+    @staticmethod
+    def get_decayed_parameter(
+        schedule: ScheduleType,
+        initial_value: float,
+        min_value: float,
+        max_step: int,
+        global_step: int,
+    ) -> float:
+        """
+        Get the value of a parameter that should be decayed, assuming it is a function of
+        global_step.
+        :param schedule: Type of learning rate schedule.
+        :param initial_value: Initial value before decay.
+        :param min_value: Decay value to this value by max_step.
+        :param max_step: The final step count where the return value should equal min_value.
+        :param global_step: The current step count.
+        :return: The value.
+        """
+        if schedule == ScheduleType.CONSTANT:
+            return initial_value
+        elif schedule == ScheduleType.LINEAR:
+            return ModelUtils.polynomial_decay(
+                initial_value, min_value, max_step, global_step
+            )
+        else:
+            raise UnityTrainerException(f"The schedule {schedule} is invalid.")
+
+    @staticmethod
+    def polynomial_decay(
+        initial_value: float,
+        min_value: float,
+        max_step: int,
+        global_step: int,
+        power: float = 1.0,
```

Comment:
There's really no way to configure this properly. Should it even be an arg?

Reply:
If at some point we want a ScheduleType.EXPONENTIAL, we do